emrichen/tags/group.py

```diff
@@ -0,0 +1,39 @@
+from collections import OrderedDict
+from sys import stderr
+
+from .loop import Loop
+from .var import Var
+
+
+class Group(Loop):
+    """
+    arguments: |
+        Accepts the same arguments as `!Loop`, except `template` is optional (default identity), plus the following:
+        `by`: (required) An expression used to determine the key for the current value
+        `result_as`: (optional, string) When evaluating `by`, the enriched `template` is available under this name.
+
+    example: TBD
+    description: Makes a dict out of a list. Keys are determined by `by`.
+    """
+    value_types = (dict,)
+
+    def enrich(self, context):
+        as_ = self.data.get('as', 'item')
+        self.data.setdefault('template', Var(as_))
```

Comment:
Yigh, feels bad to mutate `self.data`... Might bite us in the ankle at some point.

Reply:
Good point. Updated to create a copy if args need to be touched.

Even still, we need to be careful about touching `self.data` (even if it were a copy) so as to not leak state between invocations of `enrich` of the same instance of a tag as the tag instance is created only once per parse and reused (in eg. loops, same var getting `!Var`ed multiple times etc).